### PR TITLE
fix: date filter input bugs

### PR DESCRIPTION
**Pull request by nick-lanners**
## Description
This PR fixes a bug in the Date Filter where typing the date into the date inputs was incredibly buggy and difficult to use.

## Preview
**Before**
<https://amino.zonos.com/?path=/story/amino-filters--date>

**After**
<https://nick.amino.zonos.com/?path=/story/amino-filters--date>

Functionality should be the same between the two versions, but typing the dates will be more intuitive in the **After**

## Todo

- [x] Bump version and add tag

### CODE REVIEW

File: src/components/filter/filter-date/FilterDate.tsx [hunk 1]
```diff
@@ -47,12 +72,15 @@ export const FilterDate = ({
       dispatch({
         name: 'dateData.dateBegin',
         type: 'change',
-        value: value.dateBegin || null,
+        value:
+          dayjs(value.dateBegin)
+            .subtract(1, 'days')
+            .format(defaultDateFormat) || null,
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:78]:
```suggestion
        value:
          getAdjustedDate({date: value.dateBegin, type: 'begin'}),
```

**Comment by ckgardner** [comment]
Uploading Screen Recording 2024-12-30 at 10.44.19 AM.mov…

I tried typing 12/20/2024 and this happened. Appears to be a previous bug as well.

File: src/components/filter/filter-date/FilterDate.tsx [hunk 2]
```diff
@@ -37,6 +41,27 @@ export const FilterDate = ({
...
+        }
+        return date;
+      case 'is on or after':
+        return dayjs(date).subtract(1, 'days').format(defaultDateFormat);
+      case 'is before or on':
+        return dayjs(date).add(1, 'days').format(defaultDateFormat);
+      default:
+        return date;
+    }
+  };
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:63]: Looks like the return type for `dateBegin` and `dateEnd` in reducer is nullable, what do you think about making the date argument to be nullable so you don't have to have default nullish value to empty string?
Also, what do you think about make the arguments to props instead when we have more than 2 properties?
```suggestion
  const getAdjustedDate = ({date, type}: {date: string, type: 'begin' | 'end'}) => {
    switch (rangeType) {
      case 'is between':
      case 'is equal to':
        if (type === 'begin') {
          return dayjs(date).subtract(1, 'days').format(defaultDateFormat);
        }
        if (type === 'end') {
          return dayjs(date).add(1, 'days').format(defaultDateFormat);
        }
        return date;
      case 'is on or after':
        return dayjs(date).subtract(1, 'days').format(defaultDateFormat);
      case 'is before or on':
        return dayjs(date).add(1, 'days').format(defaultDateFormat);
      default:
        return date;
    }
  };
```

File: src/components/filter/filter-date/FilterDate.tsx [hunk 3]
```diff
@@ -47,12 +72,15 @@ export const FilterDate = ({
...
+        value:
+          dayjs(value.dateBegin)
+            .subtract(1, 'days')
+            .format(defaultDateFormat) || null,
       });
       dispatch({
         name: 'dateData.dateEnd',
         type: 'change',
-        value: value.dateEnd || null,
+        value: getAdjustedDate(value.dateEnd || '', 'end'),
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:83]:
```suggestion
        value: getAdjustedDate({date: value.dateEnd, type: 'end'}),
```

File: src/components/filter/filter-date/FilterDate.tsx [hunk 4]
```diff
@@ -77,12 +105,17 @@ export const FilterDate = ({
     dispatch({
       name: 'dateData.dateBegin',
       type: 'change',
-      value: editingValue.dateBegin || null,
+      value:
+        dayjs(editingValue.dateBegin)
+          .subtract(1, 'days')
+          .format(defaultDateFormat) || null,
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:111]:
```suggestion
      value: getAdjustedDate({date: value.dateBegin, type: 'begin'}),
```

File: src/components/filter/filter-date/FilterDate.tsx [hunk 5]
```diff
@@ -77,12 +105,17 @@ export const FilterDate = ({
...
     });
     dispatch({
       name: 'dateData.dateEnd',
       type: 'change',
-      value: editingValue.dateEnd || null,
+      value:
+        dayjs(editingValue.dateEnd).add(1, 'days').format(defaultDateFormat) ||
+        null,
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:118]:
```suggestion
      value: getAdjustedDate({date: value.dateEnd, type: 'end'}),
```

File: src/components/filter/filter-date/FilterDate.tsx [hunk 6]
```diff
@@ -37,6 +41,33 @@ export const FilterDate = ({
...
+        }
+        return date;
+      case 'is on or after':
+        return dayjs(date).subtract(1, 'days').format(defaultDateFormat);
+      case 'is before or on':
+        return dayjs(date).add(1, 'days').format(defaultDateFormat);
+      default:
+        return date;
+    }
+  };
```
Comment (Thienhuynh95) [inline src/components/filter/filter-date/FilterDate.tsx:69]: image
Looks like `dateEnd` is not working right 🤔

**Review by ckgardner — DISMISSED** [review]

**Review by ckgardner — APPROVED** [review]

**Pull request merged**